packages/devtools_app/lib/src/screens/memory/panes/diff/controller/simple_controllers.dart

```diff
@@ -4,7 +4,48 @@
 
 import 'package:flutter/material.dart';
 
+import '../../../../../shared/memory/adapted_heap_data.dart';
+import '../../../shared/heap/heap.dart';
+import '../../../shared/primitives/simple_elements.dart';
+import 'heap_diff.dart';
+
 class RetainingPathController {
   final hideStandard = ValueNotifier<bool>(true);
   final invert = ValueNotifier<bool>(true);
 }
+
+class ClassesTableSingleController {
+  ClassesTableSingleController({
+    required this.selection,
+    required this.heap,
+    required this.totalHeapSize,
+    required this.filterButton,
+  });
+
+  // We use functions, not [ValueListener], where we do not want widgets
+  // to subscribe for the changes, for performance reasons.
+
+  final HeapDataObtainer heap;
+  final int Function() totalHeapSize;
+  final Widget filterButton;
```

Comment:
we shouldn't be passing widgets to controller code. Widgets should only be created as part of the widget lifecycle.